torch/distributed/_tensor/_dispatch.py

```diff
@@ -431,28 +432,51 @@ def _try_replicate_spec_for_scalar_tensor(
             )
         return replication_spec
 
+    def _try_find_largest_mesh_from_args(
+        self,
+        args_list: Sequence[object],
+        kwargs: Dict[str, object],
+    ) -> Optional["DeviceMesh"]:
+        """
+        This utils finds the largest mesh from args/kwargs when implicit replication is turned on.
+        """
+        mesh: Optional[DeviceMesh] = None
+        # record the largest mesh only if implicit replication is turned on
+        for arg in chain(iter(args_list), iter(kwargs.values())):
+            if isinstance(arg, dtensor.DTensor):
+                if mesh is None or arg.device_mesh.ndim > mesh.ndim:
```

Comment:
I think the following case might not be encountered right now, but it's possible:
Let's say we have a 3D mesh, and all DTensor input are on 1D / 2D submeshes, but some 2D dtensor is on submesh indexed by [0, 1], some other 2D dtensor is on submesh indexed by [1, 2]. Then here we'd return the first 2D submesh, but it's possible that there's some 1D dtensor on submesh indexed by [2] can't get properly replicated.

I think we should check
- either: when `arg.device_mesh.ndim == mesh.ndim` they are always the same (sub)mesh; when `arg.device_mesh.ndim < mesh.ndim`, `arg.device_mesh` is a submesh of `mesh`
- or: when doing replicate, the dtensor being replicated is originally on a submesh of `mesh` which we are replicating onto.

Semantically the first makes more sense, and we should error out if there's no maximum mesh (but only maximal mesh). The efficiency comparison between the two options depends on what kind of inputs we get.